### PR TITLE
JBPM-5904: Extends test coverage for Task events

### PR DESCRIPTION
**Pull request by jakubschwan**
Tests with Task Events for not existing tasks will fail due [JBPM-5904](https://issues.jboss.org/browse/JBPM-5904)

### CODE REVIEW

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java [hunk 1]
```diff
@@ -1223,20 +1225,87 @@ public void testFindTaskEvents() throws Exception {
...
+            events = taskClient.findTaskEvents(CONTAINER_ID, taskInstance.getId(), 0, 10, SORT_BY_TASK_EVENTS_TYPE, false);
+            assertEquals(TaskEvent.TaskEventType.STOPPED.toString(), events.get(0).getType());
+            assertEquals(TaskEvent.TaskEventType.STARTED.toString(), events.get(1).getType());
+            assertEquals(TaskEvent.TaskEventType.ADDED.toString(), events.get(2).getType());
+        } finally {
+            processClient.abortProcessInstance(CONTAINER_ID, processInstanceId);
+        }
+    }
+
+    private void assertTaskEventInstance(TaskEventInstance expected, TaskEventInstance actual) {
```
Comment (triceo) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java]: If the first assertion fails, you will not get to see results of the others following it. If used, `SoftAssertions` from AssertJ would allow you to treat all these assertions as a single pass-fail.

Comment (jakubschwan) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java]: It is a good idea to use AssertJ but I think that would be better to use AssertJ for whole classes (or kie server jBPM integration tests module). If we do changes only for this PR, then we will have a few tests that use AssertJ and the other tests use JUnit and I am afraid that this could be confusing for others.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/RuntimeDataServiceIntegrationTest.java [hunk 2]
```diff
@@ -1477,6 +1477,16 @@ private TaskEventInstance getTaskEventInstanceFromListByType(List<TaskEventInsta
     }
 
     @Test
+    public void testFindTaskEventsForNotExistingTask() {
+        try {
+            taskClient.findTaskEvents(-9999l, 0, 10);
+            fail("KieServicesException should be thrown complaining about task not found.");
+        } catch (KieServicesException e) {
```
Comment (sutaakar) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/RuntimeDataServiceIntegrationTest.java]: Would be good to check also HTTP status code.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java [hunk 3]
```diff
@@ -1223,20 +1225,87 @@ public void testFindTaskEvents() throws Exception {
...
             processClient.abortProcessInstance(CONTAINER_ID, processInstanceId);
         }
     }
 
+    @Test
+    public void testFindTaskEventsForNotExistingTask() {
+        try {
+            taskClient.findTaskEvents(CONTAINER_ID, -9999l, 0, 10);
+            fail("KieServicesException should be thrown complaining about task not found.");
+        } catch (KieServicesException e) {
```
Comment (sutaakar) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java]: Would be good to check also HTTP status code.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java [hunk 4]
```diff
@@ -1223,20 +1225,87 @@ public void testFindTaskEvents() throws Exception {
 
             TaskSummary taskInstance = tasks.get(0);
 
+            TaskEventInstance expectedTaskEventInstance = TaskEventInstance
+                    .builder()
+                    .type(TaskEvent.TaskEventType.ADDED.toString())
+                    .processInstanceId(processInstanceId)
+                    .taskId(taskInstance.getId())
+                    .user(PROCESS_ID_USERTASK)
```
Comment (sutaakar) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java]: Why is user set as process id?

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java [hunk 5]
```diff
@@ -1223,20 +1225,87 @@ public void testFindTaskEvents() throws Exception {
...
+    public void testFindTaskEventsSortedBy() throws Exception {
+        Map<String, Object> parameters = new HashMap<String, Object>();
+        parameters.put("stringData", "waiting for signal");
+        parameters.put("personData", createPersonInstance(USER_JOHN));
+
+        Long processInstanceId = processClient.startProcess(CONTAINER_ID, PROCESS_ID_USERTASK, parameters);
+
+        try {
+
+            List<TaskSummary> tasks = taskClient.findTasksByStatusByProcessInstanceId(processInstanceId, null, 0, 10);
```
Comment (sutaakar) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java]: Can you please add paging test - for example to check that second page is empty

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java (hunk 3 above)
Comment (domhanak) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java]: Assertions.assertThatThrownBy(() -> taskClient.findTaskEvents(CONTAINER_ID, -9999l, 0, 10))
                .isInstanceOf(KieServicesException.class)
                .hasMessageContaining("No task found with id");

As a point to counter your statement below that it would be confusing for others to employ AssertJ here.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java (hunk 1 above)
Comment (domhanak) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java]: I think that by replacing org.junit.Assert in whole class isn't really a problem and would greatly enhance readability of this test class.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java (hunk 5 above)
Comment (jakubschwan) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java]: Paging added into sorting tests.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/RuntimeDataServiceIntegrationTest.java (hunk 2 above)
Comment (jakubschwan) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/RuntimeDataServiceIntegrationTest.java]: Add check of the HTTP status code for REST configuration.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java (hunk 4 above)
Comment (sutaakar) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java]: it is because added task is a system operation, logged system user who performs the operation

**Comment by mswiderski** [comment]
@tsurdilo assigned this PR to you as you work on that jira so please include this in your work

**Comment by tsurdilo** [comment]
@jakubschwan @mswiderski 
PRs for this Jira: 
https://github.com/kiegroup/droolsjbpm-integration/pull/933
https://github.com/kiegroup/jbpm/pull/836

@jakubschwan you can go ahead and close this PR as all changes here are included in https://github.com/kiegroup/droolsjbpm-integration/pull/933

Thanks!

**Comment by mswiderski** [comment]
closing since it's already merged with https://github.com/kiegroup/droolsjbpm-integration/pull/933
